pywr/solvers/cython_glpk.pyx

```diff
@@ -677,6 +723,29 @@ cdef class CythonGLPKSolver(GLPKSolver):
         self.stats['objective_update'] += time.perf_counter() - t0
         t0 = time.perf_counter()
 
+        for agg_node in self.aggregated_with_factor_params:
+            factors = agg_node.get_factors(scenario_index)
+
+            # normalise factors
+            f0 = factors[0]
+            factors_norm = [f0/f for f in factors]
+
+            row = agg_node.__agg_factor_data.row
+            indptr_array = agg_node.__agg_factor_data.ind_pntr
+
+            for n in range(len(agg_node.nodes)-1):
+
+                ind = agg_node.__agg_factor_data.ind[indptr_array[n]:indptr_array[n+1]]
+                val = agg_node.__agg_factor_data.val[indptr_array[n]:indptr_array[n+1]]
+                length = len(ind) - 1
+
+                # only update factors for second node of each row
+                for i in range(agg_node.__agg_factor_data.node_ind, val.shape[0]):
+                    val[i] = -factors_norm[n+1]
+
+                set_mat_row(self.prob, row+n, length, &ind[0], &val[0])
```

Comment:
This is definitely more how I envisaged this. No re-allocation during the timesteps.